libs/vkd3d/swapchain.c

```diff
@@ -873,8 +918,23 @@ static HRESULT STDMETHODCALLTYPE dxgi_vk_swap_chain_Present(IDXGIVkSwapChain *if
     request->dxgi_hdr_metadata = chain->user.dxgi_hdr_metadata;
     request->modifies_hdr_metadata = chain->user.modifies_hdr_metadata;
     request->begin_frame_time_ns = chain->user.begin_frame_time_ns;
+    request->low_latency_frame_id = chain->queue->device->frame_markers.present;
     chain->user.modifies_hdr_metadata = false;
 
+    if (chain->queue->device->vk_info.NV_low_latency2)
+    {
+        pthread_mutex_lock(&chain->present.low_latency_state_update_lock);
+        request->requested_low_latency_state = chain->requested_low_latency_state;
```

Comment:
Why is a lock taken here? DXGI side takes care of locking the Present() call.

Reply:
I added the low_latency_state_update_lock to ensure that a call to Present couldn't race with a call to SetLatencySleepMode.